File: Leetcode/1. Two Sum/solution4.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        h = {}
        for index, num in enumerate(nums):
            if num in h: return [h[num], index]
            h[target - num] = index
        raise ValueError('No two sum solution')

File: Leetcode/1. Two Sum/solution4.cpp
class Solution {
public:
    vector<int> twoSum(vector<int> &nums, int target) {
        vector<int> map(2048, -1);

        for (int i = 0; i < nums.size(); i++) {
            if (map[(target - nums[i]) & 2047] != -1) {
                return {map[(target - nums[i]) & 2047], i};
            }

            map[nums[i] & 2047] = i;
        }

        throw invalid_argument("No two sum solution");
    }
};